IDLE_DWELL/Find_Idle_Dwells.py

```diff
@@ -0,0 +1,256 @@
+                 (state_dict["present_fp_instrument"] == "ACIS-S") or \
+                 (state_dict["radmon_status"] == "RADMON_DIS") ) and \
+                 (dwell_time_secs >= long_dwell_cutoff):
+                dwell_string = " ".join((">>> Warning - Long Dwell:", state_dict["present_fp_instrument"], "in the F.P.",  state_dict["radmon_status"], "Dwell time: ", str(round(dwell_time_secs,1)), "seconds"))
+
+                # Print the warning out to STDOUT so that the load reviewer sees it
+                # and it gets logged in the log file we create when running lr
+                print( dwell_string)
+                
+                # Append the dwell stop date and time, and the dwell string to
+                # the list of dwell strings.  This will cause the information line
+                # to appear just before the Start Science
+                long_dwell_list.append([dwell_stop_date, dwell_stop_time, dwell_string])
+                
+            # Set the value of was_an_obs to True
+            state_dict["was_an_obs"] = True
+            
+            # Set AAOO flag to False so that we will process the first AA00
+            # command that comes after this Start Science when we get to it
+            state_dict["first_AA00_flag"] = False
+ 
+         # Check for a Stop Science: AA00000000
+         if (stop_science_command in sim_acis_commands[index]["tlmsid_string"]) and \
+           (state_dict["first_AA00_flag"] == False):
+            
+            # --------------- DWELL START: AA00 ----------------------------------------------
+            # You found the  Stop Science for that observation. Capture the
+            # time of that command as the dwell start date and time
+            dwell_start_date = sim_acis_commands[index]["date"]
+            dwell_start_time = sim_acis_commands[index]["time"]
+
+            # Copy the present FP intrument into the previous FP instrument
+            state_dict["previous_fp_instrument"] = state_dict["present_fp_instrument"] 
+
+            # Set the AA00 flag to True so that no AA00 command between this
+            # one and the next Start Science will be processed
+            state_dict["first_AA00_flag"] = True
+ 
+         # Haven't seen any Start Science commands, keep looking
+         index += 1
+
+    # Done finding any long dwells. If there are lines in long_dwell_list, insert
+    # them in a copy of ACIS_LoadReview.txt called ACIS_LoadReview.txt.DWELL_COMMENT
+
+    if len(long_dwell_list)> 0:
+        icia.Insert_Comment_In_ALR(long_dwell_list, load_week_path, "DWELL_COMMENTS")
+
+
+    # Copy the updated ACIS-LoadReview.txt file
+    # If the test flag was False, then move the .DWELL_COMMENTS file to
+    # ACIS-LoadReview.txt.
+    # If it was True then we leave the original ACIS-LoadReview.txt and the
+    # ACIS-LoadReview.txt.ERRORS files intact for comparison.
+    if dwell_args.test == False:
```

Comment:
same as above here

Reply:
Also disagree here because if you print out the actual value of dwell_args.test (assuming the user didn't include the argument) you'd get the exact value of False.

In general I much prefer to test on the actual value one can get vs using a test, like "not",  where there could be numerous inputs that give the same result (same as above: x = [] not x returns True )